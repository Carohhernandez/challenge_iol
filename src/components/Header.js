import React from 'react';
import { Link } from "react-router-dom";
import styled from 'styled-components';

// SVG
import {ReactComponent as RickLogo} from '../svg/rick.svg';

const HeaderContainer = styled.div`
    display: flex;
    align-items: center;

    background-color: #fff;
    padding: 30px 20px; 
`;

const HeaderLogo = styled(RickLogo)`
    width: 80px;
    height: 80px;
`;

const HeaderTitle = styled.h1`
    color: #000;
    padding-left: 20px;
`;

const Header = () => (
    <HeaderContainer>
        <Link to="/"><HeaderLogo /></Link>
        <HeaderTitle>The Rick and Morty Web App</HeaderTitle>
    </HeaderContainer>
);

export default Header;