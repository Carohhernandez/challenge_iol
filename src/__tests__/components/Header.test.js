import { render, cleanup } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom';
import Header from '../../components/Header';

afterEach(cleanup);

test("renders Header correctly", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <Header />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});
