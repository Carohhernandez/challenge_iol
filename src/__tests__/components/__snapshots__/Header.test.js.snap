// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe bCQUeM"
  >
    <a
      href="/"
    >
      <svg
        class="sc-gswNZR CfbSo"
      >
        rick.svg
      </svg>
    </a>
    <h1
      class="sc-dkrFOg iplnsQ"
    >
      The Rick and Morty Web App
    </h1>
  </div>
</DocumentFragment>
`;
